Source/WebInspectorUI/UserInterface/Views/SpreadsheetCSSStyleDeclarationSection.js

```diff
@@ -694,6 +727,103 @@ WI.SpreadsheetCSSStyleDeclarationSection = class SpreadsheetCSSStyleDeclarationS
 
         this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.FilterApplied, {matches});
     }
+
+    _handleSpreadsheetSelectorFieldStartedEditing(event)
+    {
+        this._headerElement.classList.add("editing-selector");
+    }
+
+    _handleSpreadsheetSelectorFieldStoppedEditing(event)
+    {
+        this._headerElement.classList.remove("editing-selector");
+    }
+
+    _handleSpreadsheetSelectorFieldDidCommit(changed)
+    {
+        let selectorText = this._selectorElement.textContent.trim();
+        if (selectorText && changed) {
+            this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.SelectorWillChange);
+            this._style.ownerRule.setSelectorText(selectorText).finally(this._renderSelector.bind(this));
+        } else
+            this._discardSelectorChange();
+    }
+
+    _handleSpreadsheetSelectorFieldWillNavigate(direction)
+    {
+        if (direction === "forward")
+            this._propertiesEditor.startEditingFirstProperty();
+        else if (direction === "backward") {
+            for (let i = this._groupingElements.length - 1; i >= 0; ++i) {
+                let groupingElementTextField = this._groupingElements[i].associatedTextField;
+                if (groupingElementTextField) {
+                    groupingElementTextField.startEditing();
+                    return;
+                }
+            }
+            if (this._delegate.spreadsheetCSSStyleDeclarationSectionStartEditingAdjacentRule) {
+                const delta = -1;
+                this._delegate.spreadsheetCSSStyleDeclarationSectionStartEditingAdjacentRule(this, delta);
+            } else
+                this._propertiesEditor.startEditingLastProperty();
+        }
+    }
+
+    _handleSpreadsheetGroupingFieldDidCommit(textField, changed)
+    {
+        let groupingTextElement = textField.element;
+        let text = groupingTextElement.textContent.trim();
+
+        if (!changed || !text) {
+            this._handleSpreadsheetGroupingFieldDidDiscard(textField);
+            return;
+        }
+
+        this.dispatchEventToListeners(WI.SpreadsheetCSSStyleDeclarationSection.Event.SelectorWillChange);
+        groupingTextElement.representedGrouping.setText(text).catch((error) => {
+            console.error(error);
```

Comment:
is this necessary?  it should already be caught by the uncaught exception reporter.  also if truly so, please use `WI.reportInternalError`